SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/TypeStructure.cpp

```diff
@@ -12,21 +12,30 @@ using namespace SpatialGDKEditor::Schema;
 
 TArray<EReplicatedPropertyGroup> GetAllReplicatedPropertyGroups()
 {
-	return { REP_MultiClient, REP_SingleClient, REP_InitialOnly };
+	static TArray<EReplicatedPropertyGroup> ReplicatedPropertyGroups = []() {
+		TArray<EReplicatedPropertyGroup> Temp;
+		for (uint32 i = EReplicatedPropertyGroup::REP_First; i < EReplicatedPropertyGroup::REP_Count; i++)
+		{
+			Temp.Add(static_cast<EReplicatedPropertyGroup>(i));
+		}
+		return Temp;
+	}();
+
+	return ReplicatedPropertyGroups;
 }
 
 FString GetReplicatedPropertyGroupName(EReplicatedPropertyGroup Group)
 {
-	if (Group == REP_SingleClient)
+	switch (Group)
 	{
+	case REP_SingleClient:
 		return TEXT("OwnerOnly");
-	}
-	else if (Group == REP_InitialOnly)
-	{
+	case REP_InitialOnly:
 		return TEXT("InitialOnly");
-	}
-	else
-	{
+	case REP_ServerOnly:
+		return TEXT("ServerOnly");
+	default:
+		static_assert(REP_Count == 4, "Unexpected number of ReplicatedPropertyGroups, please update this function.");
```

Comment:
nit - here and elsewhere - I'd put these static asserts prior to the beginning of the switch statement